### ISSUE TITLE
Strange import optimization behaviour of in-file modules (`mod {}`)

### ISSUE THREAD

**Issue by ConcurrentMarxistGC** [intellij-rust/intellij-rust#2358]
## Version
* ~~0.2.0.2086 (apologies if this is patched already)~~
* 0.2.0.2089-173 (tested with new version, bug is still here)

## Description
* The import optimization system (code formatter) moves `use` statements outside of a `mod {}` block.
* The import optimization system does not format `use` statements inside a `mod {}` block

## Use cases
When tests code should be contained inside the parent module file to prevent a mess of files, in-file module declarations are used and usually they import some code to work the tests.

#### Original code example
```rust
use std::vec::Vec;

fn is_bug_report() -> bool { true }

#[cfg(test)]
mod tests {
	use std::cmp::PartialEq;
	use super::*;

	#[test]
	fn my_test() {
		assert_eq!(is_bug_report(), true);
	}
}
```

#### After formatting
Use statements are moved above, outside the module declaration
```rust
use std::vec::Vec;

fn is_bug_report() -> bool { true }

use std::cmp::PartialEq;
use super::*;

#[cfg(test)]
mod tests {
	#[test]
	fn my_test() {
		assert_eq!(is_bug_report(), true);
	}
}
```

#### Expected behaviour
1. The use statements are not moved outside of the module declaration.
2. See the next related bug

### Module's `use`s are not sorted/optimized
When formatted, the following code is not changed by the plugin at all
```rust
fn is_bug_report() -> bool { true }

#[cfg(test)]
mod tests {
	use super::*;
	use std::cmp::PartialEq;

	#[test]
	fn my_test() {
		assert_eq!(is_bug_report(), true);
	}
}
```

#### Expected behaviour
The use statements are swapped like this
```rust
use std::cmp::PartialEq;
use super::*;
```

**Comment by ConcurrentMarxistGC** [comment]
Bump.

**Comment by krampenschiesser** [comment]
Same here, my test imports are moved up too

**Comment by Undin** [comment]
Fixed in #2521
